ddtrace/tracer/baggage_test.go

```diff
@@ -0,0 +1,136 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024 Datadog, Inc.
+
+package tracer
+
+import (
+	"context"
+	"testing"
+)
+
+func TestBaggageFunctions(t *testing.T) {
+	t.Run("SetBaggage and Baggage", func(t *testing.T) {
+		ctx := context.Background()
+
+		// Set a key/value in the baggage
+		ctx = SetBaggage(ctx, "foo", "bar")
+
+		// Retrieve that value
+		got, ok := Baggage(ctx, "foo")
+		if !ok {
+			t.Error("Expected key \"foo\" to be found in baggage, got ok=false")
+		}
+		if got != "bar" {
+			t.Errorf("Baggage(ctx, \"foo\") = %q; want \"bar\"", got)
+		}
+
+		// Ensure retrieving a non-existent key returns an empty string
+		got, ok = Baggage(ctx, "missingKey")
+		if ok {
+			t.Error("Expected key \"missingKey\" to not be found, got ok=true")
+		}
+		if got != "" {
+			t.Errorf("Baggage(ctx, \"missingKey\") = %q; want \"\"", got)
+		}
```

Comment:
Good differentiating between unset key vs a key with an empty string value with the above test condition!